platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsToolingApiGradleProjectIntegrationTest.groovy

```diff
@@ -201,4 +201,59 @@ class IsolatedProjectsToolingApiGradleProjectIntegrationTest extends AbstractIso
         fixture.assertStateLoaded()
     }
 
+    def "root GradleProject model is invalidated when a child project configuration changes"() {
```

Comment:
💡 Ideally, the root GradleProject model would only be invalidated when a consumed child project model changes. That sort of dependency tracking is what we expect to have with _Shared Data_ so perhaps some of these models would be migrated to shared data once the time is ripe.